### ISSUE TITLE
Redesign calendar sharing

### ISSUE THREAD

**Issue by ChristophWurst** [nextcloud/calendar#4124]
### Is your feature request related to a problem? Please describe.

Calendar sharing is currently a bit problematic from a usability and accessibility point of view, so it was added to https://github.com/nextcloud/calendar/issues/3540.

### Describe the solution you'd like

> "can edit" is difficult to uncheck with keyboard, as the focus switches somehow after checking it

To be defined by designers.

### Describe alternatives you've considered

To be defined by designers.

**Comment by rakekniven** [comment]
Today I got a message on Transifex that the russian translation for "can edit" takes to much space in width and the username is nearly unreadable.

So I checked the russian translations and they are unified and valid.

On my NC server I selected an test account with german lanugage. See following screenshot.
Translation is "kann bearbeiten" and my username is shortened. Usernames with more than 4 characters will be unreadable as well. 

<img width="742" alt="Screenshot 2022-05-06 at 11 24 52" src="https://user-images.githubusercontent.com/2069590/167110700-1596f2b4-977f-4bb6-819f-466b0d337e95.png">

And there are more languages with longer translations like german e. g. russian, brasilian, croatian, estonian ...
See https://www.transifex.com/nextcloud/search/?q=project%3Anextcloud+source_text%3A%27can+edit%27 

From my POV the UI needs to be adapted.
Ideas:
1: Put permission checkboxes below username
2: Allow user to increase size of left sidebar using a slider or something similar. Currently the size of the sidebar is fixed, or?

**Comment by jancborchardt** [comment]
Some options here where to move the sharing:
- Into an ActionMenu: Barely an improvement, but since calendars are not often shown and it’s in place it might be a good small enhancement
- Into a modal: Sure, always works, and is a generic fix where we can also show if it’s shared with many people
- Right sidebar: Seems confusing in calendar actually, cause the sidebar is really dedicated to event details. Seems strange to open it for calendar details, especially also because it opens something on the right side for an action you took on the left.

What seems most fitting is if we put "Calendar details & sharing" into a modal together, so name, color, and sharing can be edited in one place. Export and Delete calendar can be done there as well, basically just like the "Conversation settings" in Talk.

What do you think @nimishavijay?

**Comment by nimishavijay** [comment]
Sounds good! In the action menu we could have "Details" and "Sharing" which could open different sections of the same settings modal, and "Copy private link" and "Delete"/"Unshare from me".
The delete option could also be available inside the modal.

**Comment by jancborchardt** [comment]
@nimishavijay yup, sounds good!

**Comment by rakekniven** [comment]
I can't really imagine it yet (I'm not that familiar with the wording).

Does this solve the issue I described above?

**Comment by nimishavijay** [comment]
@rakekniven The general idea is to have the sharing in a modal, not in the navigation. The modal would have more space so yes, the words won't be cut off even if they are long :)

**Comment by ChristophWurst** [comment]
With Nextcloud 25 design the sharing design got even worse: 

![Bildschirmfoto vom 2022-09-13 14-34-36](https://user-images.githubusercontent.com/1374172/189904495-4d6e0dea-7ec5-4118-9675-ceb86cf231eb.png)

We'll see if we can move to the new modal flow now.

Ref https://github.com/nextcloud/calendar/pull/4491

**Comment by st3iny** [comment]
Additional enhancements discussed during our latest design reviews:
* [ ] Show "Shared with you by ..." in the three dot menu at the top (if shared)
  * [ ] Icon should be the avatar
* [ ] Add two dedicated entries to the three dot menu: 1. sharing and 2. details
  * [ ] Open the same modal but scroll to different sections
  * [ ] Share button should be on top

**Comment by ChristophWurst** [comment]
I don't remember why this is labeled *blocked* so I'm removing the status.
